### PR TITLE
[18.0][MIG] auditlog: Migration to 18.0 

### PR DESCRIPTION
**Pull request by lembregtse**

### CODE REVIEW

**Comment by gurneyalex** [comment]
Warning, there is an issue on 17.0 resulting in some workers not implementing the correct logging. 

https://github.com/OCA/server-tools/pull/3088  is supposed to fix it.

**Comment by StefanRijnhart** [comment]
@lembregtse Can you update your branch to include #3088 which is now merged?

**Comment by mylbco** [comment]
@lembregtse any update on this ;-)

**Comment by lembregtse** [comment]
@StefanRijnhart @mylbco FYI, I've redone the PR. I somehow missed your messages!

**Comment by StefanRijnhart** [comment]
@lembregtse thanks! There is a fixup commit involved in that last merge. This would be the right moment to squash it if you would be so kind.

**Comment by lembregtse** [comment]
@StefanRijnhart misunderstood, but squashed now!

**Comment by StefanRijnhart** [comment]
@lembregtse Thanks, but now I'm confused about the message in the migration commit.
image
Was something squashed here? What change in the commit prevents empty logs? Is it related to https://github.com/OCA/server-tools/pull/3137?

File: auditlog/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
+# Copyright 2015 ABF OSIELL <https://osiell.com>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+{
+    "name": "Audit Log",
+    "version": "18.0.1.0.1",
```
Comment (StefanRijnhart) [inline auditlog/__manifest__.py]:
```suggestion
    "version": "18.0.1.0.0",
```

**Comment by lembregtse** [comment]
@StefanRijnhart my bad, I seem to have "Ctrl+R'ed" into a commit message I used back then to try (as you mentioned) something locally for Raf's fix in the other PR.

I've removed the message from the commit, it was not relevant here, it was for a squash I did to fix the cron data file (removing doall and numbercall).

I've changed the version number as well!

**Review by StefanRijnhart — APPROVED** [review]
Thanks for the update, looking even better than before!

**Comment by ristecona** [comment]
if possible please merge, thanks a lot

**Comment by StefanRijnhart** [comment]
@ristecona A technical of functional review would be appreciated!

**Comment by MiquelRForgeFlow** [comment]
@lembregtse please cherry-pick https://github.com/OCA/server-tools/pull/3147

**Comment by StefanRijnhart** [comment]
And https://github.com/OCA/server-tools/pull/3137

**Comment by lembregtse** [comment]
@StefanRijnhart I've cherrypicked both commits, fascinatingly it fails in the pipeline, didn't locally. Difference seems to be the "phonenumbers" python package (I did not have it installed locally). Checking it out.

**Comment by lembregtse** [comment]
It makes sense, since the phone field was excluded, with @rven changes the test-case should have failed in 17.0 as well as it normally should not log any auditlog. I added an extra "changed field" to restore the test-cases original purpose. But I do not understand why installing the phonenumbers makes any difference in how auditlog "logs" stuff. Because I believe this should have failed in 17.0 as well.

**Review by StefanRijnhart — COMMENTED** [review]
Thanks for including the most recent changes! You are right, we would expect the tests to fail on 17 as well (they did for me locally) but in some cases the field `phone_sanitized` is written in the same call. I picked your fix in https://github.com/OCA/server-tools/pull/3153.
Can you therefore align the commit message of the fixing commit to `[FIX] auditlog: adapt exclude fields test to #3137`?

**Comment by lembregtse** [comment]
@StefanRijnhart done!

**Review by StefanRijnhart — APPROVED** [review]
Thanks!

**Comment by miikanissi** [comment]
Can this get merged?

**Comment by StefanRijnhart** [comment]
@miikanissi the rules here are:
> The code will be merged upon 3 positive reviews within 5 days (or 2 after more than 5 days). At least one of the review above must be from a member of the [PSC](https://odoo-community.org/page/List) or someone from the [OCA Core Maintainer](https://github.com/orgs/OCA/teams/core-maintainers).

As mine is the only review so far, this PR does not qualify for merging yet. Your review can help! At this point, a functional test would make most sense I think. You can test using this PRs runboat container, see https://odoo-community.org/resources/review. If you are familiar with this module's code, a code review is fine too, of course.

**Review by miikanissi — APPROVED** [review]
LGTM - did a functional test with multiple different rulesets and models.

**Pull request merged**